pkg/util/instancemetadata/fake.go

```diff
@@ -0,0 +1,30 @@
+package instancemetadata
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"os"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// NewFake returns InstanceMetadata for use in unit tests, without validating that env vars are set
+func NewFake() (InstanceMetadata, error) {
+	environment := azure.PublicCloud
+	if value, found := os.LookupEnv("AZURE_ENVIRONMENT"); found {
+		var err error
+		environment, err = azure.EnvironmentFromName(value)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &instanceMetadata{
+		tenantID:       os.Getenv("AZURE_TENANT_ID"),
+		subscriptionID: os.Getenv("AZURE_SUBSCRIPTION_ID"),
+		location:       os.Getenv("LOCATION"),
+		resourceGroup:  os.Getenv("RESOURCEGROUP"),
+		environment:    &environment,
+	}, nil
```

Comment:
This code looks identical to that found in [`pkg/util/instancemetadata/dev.go`](https://github.com/Azure/ARO-RP/blob/d2e6fcbcba9900740e76efe0a3b508d2bac5fc87/pkg/util/instancemetadata/dev.go#L25-L40). Any reasons why they can't be DRY up?

Not sure how this unit test is set up, but generally, I try to avoid introducing external dependencies like env vars in unit tests, to avoid potential flakiness. Where are these env vars being set?

Reply:
I will hold off on refactoring to avoid repetition for now because this entire approach might be ill-advised.

To explain the purpose of this, in a couple unit tests, there is a new dependency on `InstanceMetadata`. Currently, the simplest way of making one of these from outside the `instancemetadata` package requires that env vars be set, which is not the case when running unit tests.

Reply:
Removing this and using `MockInstanceMetadata` as should have been the obvious move :facepalm: